### PR TITLE
Add new described/urls section to the definition

### LINKED ISSUE #403: Add new described/urls section to the definition
We want to surface a set of URLs in the definition themselves, to be computed by the service.

for github:

```json
{
  "described": {
      "urls": {
           "homepage": "https://github.com/clearlydefined/service",
           "component": "https://github.com/clearlydefined/service/tree/9f6f79a17460c92fb3876d9f485411d43aa70374",
           "download": "https://github.com/clearlydefined/service/archive/9f6f79a17460c92fb3876d9f485411d43aa70374.zip"
       }
  }
}
```

for npm:

```json
{
  "described": {
      "urls": {
           "homepage": "https://www.npmjs.com/package/tiny-attribution-generator",
           "component": "https://www.npmjs.com/package/tiny-attribution-generator/v/0.0.6",
           "download": "https://registry.npmjs.com/tiny-attribution-generator/tiny-attribution-generator-0.0.6.tgz"
       }
  }
}

### PR DESCRIPTION
**Pull request by storrisi**
Fixes #403

### CODE REVIEW

**Comment by storrisi** [comment]
@dabutvin this could now be reviewed.
I have applied the methods in the `definitionService` only, as per your guideline.

File: business/definitionService.js [hunk 1]
```diff
@@ -500,6 +504,14 @@ class DefinitionService {
     }
   }
 
+  async _ensureUrls(coordinates, definition) {
+    const homePageUrl = buildHomepageUrl(coordinates)
+    const componentUrl = buildComponentUrl(coordinates)
```
Comment (dabutvin) [inline business/definitionService.js]: see https://github.com/clearlydefined/service/issues/403#issuecomment-465585617 from @pombredanne 

I agree, we should come up with a better naming here


here's an updated proposal

```json
{
  "described": {
      "urls": {
           "registry": "https://www.npmjs.com/package/tiny-attribution-generator",
           "version": "https://www.npmjs.com/package/tiny-attribution-generator/v/0.0.6",
           "download": "https://registry.npmjs.com/tiny-attribution-generator/tiny-attribution-generator-0.0.6.tgz"
       }
  }
}
```

```json
{
  "described": {
      "urls": {
           "registry": "https://github.com/clearlydefined/service",
           "version": "https://github.com/clearlydefined/service/tree/9f6f79a17460c92fb3876d9f485411d43aa70374",
           "download": "https://github.com/clearlydefined/service/archive/9f6f79a17460c92fb3876d9f485411d43aa70374.zip"
       }
  }
}
```

File: business/definitionService.js [hunk 2]
```diff
@@ -272,9 +275,10 @@ class DefinitionService {
     set(definition, 'licensed.score', licensedScore)
   }
 
-  _finalizeDefinition(coordinates, definition) {
+  async _finalizeDefinition(coordinates, definition) {
     this._ensureFacets(definition)
     this._ensureSourceLocation(coordinates, definition)
+    await this._ensureUrls(coordinates, definition)
```
Comment (dabutvin) [inline business/definitionService.js]: 👍

File: business/definitionService.js (hunk 1 above)
Comment (storrisi) [reply business/definitionService.js]: Makes sense for me, i'm going to update this

**Comment by storrisi** [comment]
@dabutvin i've update the names following your comment and updated the schema as well.

**Pull request merged**